src/core/server/models/user/user.ts

```diff
@@ -468,7 +468,7 @@ function hashPassword(password: string): Promise<string> {
   return bcrypt.hash(password, 10);
 }
 
-export type InsertUserInput = Omit<
+export type FindOrCreateUserInput = Omit<
```

Comment:
I approve of the fact that you renamed this to make it match the functionality it was actually doing.

However, I am not a fan of functions that do multiple things. It tends to violate single responsibility principle.

Particularly, in this case its name implies both returning a user via "Find" and also updating/creating via "CreateUser".

I feel these could maybe be broken into two separate operation tasks and the multi-operation function calls each individually? This adds difficulty, as it's not as atomic (hinders race condition handling) but it also prevents what could be a code smell developing.